echo-pipelinetriggers/src/main/java/com/netflix/spinnaker/echo/pipelinetriggers/eventhandlers/ManualEventHandler.java

```diff
@@ -93,6 +101,19 @@ public ManualEvent convertEvent(Event event) {
     }
   }
 
+  @Override
+  public List<Pipeline> getMatchingPipelines(ManualEvent event, PipelineCache pipelineCache) throws TimeoutException {
+    if (!isSuccessfulTriggerEvent(event)) {
+      return Collections.emptyList();
+    }
+
+    return pipelineCache.getPipelinesSync().stream()
```

Comment:
manual events still go over the entire pipeline collection (cc @marchello2000)

Reply:
someday, maybe we will have `triggerType = [manual, git, pipeline, etc]` and `invocationType = [manual/etc]` and we will be able to avoid this iteration. For now, it makes sense